### PR TITLE
Feature contributors file

### PR DESCRIPTION
**Pull request by daidoji**
Rough draft of potential CONTRIBUTORS.md file for discussion/review.

### CODE REVIEW

**Review by m00sey — COMMENTED** [review]
Should this be "contributing.md" I would expect contributors to be a list of people.

File: Contributors.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+However, this particular file is more about contributing code.  The general principal in this (and all open source really) is if you can think of a way that things might be better, instead of just suggesting it or discussing it, often times its a good idea to do some work to show what/how/when an idea might look like or contributing code/scripts/tests/issues.  Code >> discussion 9 times out of ten.  Balanced of course with the caveat not to go off into the ivory tower for months and do lots of work before presenting your work and finding out that the community doesn't like it.  Small prototypes/examples are often best.  The rest of the details of contributing code are below.
+
+## Contribution Guidelines
+
+### Code Style/Conventions
+[PythonStyleGuide](./ref/PythonStyleGuide.md)  
+
+### Commit Message Guidelines
+There are no hard and fast guidelines (TODO: is this true?) but it is helpful to:
```
Comment (m00sey) [inline Contributors.md:31]: I mean we don't have rules, but a descriptive to the point message is ideal.

"Adds selective disclosure mechanism"
"Updates test coverage for coring"

We should discuss if we want to start requiring sign off.

File: Contributors.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+7. Iterate 2-6 if your change needs some fixes/updates
+
+## Reporting Bugs or Requesting Features
+- **SEARCH FOR THE BUG OR FEATURE IN THE CURRENT ISSUES IN REPO**.  If it already exists, add a comment/script/test/+1/whatever there.  Duplicates BAD.
+- If the bug/feature doesn't exist create an issue wherein you describe the bug, feature, or issue with as much detail as possible (but maybe not enough that you overload the reader with details).
+- Code snippets, scripts, or test cases should be added to the issue if possible.  It helps with saving maintainers time and can drastically speed the development process.
+- Message in the appropriate discord channel to let people know about your bug/feature/issue, but remember that maintainers maintain at their own pace and discretion on issues of their choosing.  Its best not to ping them more than once a week.  As with all open source, if its an ultra critical bug/feature for you, cash bounties certainly incentivize people to pay attention and offer to help you directly.
+
+## Code of Conduct and Respect
+Treat people with respect.  (TODO: Should we add a more complex code of conduct?)
```
Comment (m00sey) [inline Contributors.md:58]: We have something in discord, we could look at that

**Review by m00sey — CHANGES_REQUESTED** [review]
Requesting changes so we can either open new issues or get clarity

**Comment by m00sey** [comment]
Thanks for the PR!

**Comment by daidoji** [comment]
> Should this be "contributing.md" I would expect contributors to be a list of people.

Oh whoops, yeah probably a good call.  I will rename.

**Comment by daidoji** [comment]
@m00sey renamed and addressed your points.  Def wanted changes and input from everyone else to provide clarity which is why its in Draft form.  Hopefully, we can discuss it tomorrow.

File: Contributing.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+### Commit Message Guidelines
+There are no hard and fast guidelines but it is helpful to:
+- Descriptive, to the point messages are ideal
+- Link to the issue you're trying to solve with a message of how the commit does or doesn't solve that issue.
+- Be sure to note important issues for checkpoint commits like "does not build"
+- Most importantly, the commit message should explain the WHY of all the code changes.  When reviewing, reviewers will be confused if code changes for things you didn't list in the commit aren't there.
+
+### Branching Strategy
+Git flow.  Branch from development, name your feature branches something like `feature-name-of-my-feature` and bugs something like `bug-bug-name` and link the github issue that you (or someone else) def should have created for most non-trivial bugs.  When you have fully implemented or fixed, submit a PR to WebOfTrust/keripy.  If you need feedback it can also be appropriate to submit a **draft** PR to this repository.
```
Comment (m00sey) [inline Contributing.md:38]: I don't think we need to bring GitFlow into it, I think your description is enough.

**Review by m00sey — APPROVED** [review]
not a blocking comment

File: Contributing.md (hunk 3 above)
Comment (daidoji) [reply Contributing.md:38]: Good point.  Will take it out.

File: Contributing.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# Contributors' Guide
+
+## Introduction
+
+Welcome to keripy's contribution guide! This project aims to develop a reference implementation of the keri standards and protocols developed in WebOfTrust on github. We deeply appreciate the time and effort of contributors like you!
```
Comment (m00sey) [inline Contributing.md:5]: s/keri/KERI

**Review by m00sey — CHANGES_REQUESTED** [review]

**Comment by pfeairheller** [comment]
@SmithSamuelM recommendation from 10/19/2023 Dev Meeting to move the licensing section to the top (I suggest under Introduction)

**Review by m00sey — APPROVED** [review]

**Pull request merged**